src/NUnitEngine/nunit.engine/TestEngineResult.cs

```diff
@@ -152,5 +151,25 @@ public void Add(XmlNode xml)
         }
 
         #endregion
+
+        public static void Serialize(TestEngineResult value, BinaryWriter writer)
+        {
+            writer.Write(checked((byte)value._xmlText.Count));
+
+            foreach (var text in value._xmlText)
+                writer.Write(text);
+        }
+
+        public static TestEngineResult Deserialize(BinaryReader reader)
+        {
+            var value = new TestEngineResult();
+
+            var count = reader.ReadByte();
+
+            for (var i = 0; i < count; i++)
+                value._xmlText.Add(reader.ReadString());
+
+            return value;
+        }
```

Comment:
Should this class still have the Serializable/NonSerialized attributes, now we're serializing 'by hand'?

Reply:
Yes, so that remoting can still move this between AppDomains for the .NET Framework build's AppDomain runner. (https://github.com/nunit/nunit-console/pull/576#discussion_r266556021)